gitman/git.py

```diff
@@ -15,51 +17,67 @@
 
+def gitsvn(*args, **kwargs):
+    return call('git', 'svn', *args, **kwargs)
+
+def clean_dir(path):    
+    for root, dirs, files in os.walk(path):
+        for f in files:
+            os.unlink(os.path.join(root, f))
+        for d in dirs:
+            shutil.rmtree(os.path.join(root, d))
 
-def clone(repo, path, *, cache=settings.CACHE, sparse_paths=None, rev=None):
+def clone(type, repo, path, *, cache=settings.CACHE, sparse_paths=None, rev=None):
     """Clone a new Git repository."""
     log.debug("Creating a new repository...")
 
-    name = repo.split('/')[-1]
-    if name.endswith(".git"):
-        name = name[:-4]
-
-    reference = os.path.join(cache, name + ".reference")
-    if not os.path.isdir(reference):
-        git('clone', '--mirror', repo, reference)
-
-    normpath = os.path.normpath(path)
-    if sparse_paths:
-        os.mkdir(normpath)
-        git('-C', normpath, 'init')
-        git('-C', normpath, 'config', 'core.sparseCheckout', 'true')
-        git('-C', normpath, 'remote', 'add', '-f', 'origin', reference)
-
-        with open("%s/%s/.git/info/sparse-checkout" % (os.getcwd(), normpath), 'w') as fd:
-            fd.writelines(sparse_paths)
-        with open("%s/%s/.git/objects/info/alternates" % (os.getcwd(), normpath), 'w') as fd:
-            fd.write("%s/objects" % reference)
-
-        # We use directly the revision requested here in order to respect,
-        # that not all repos have `master` as their default branch
-        git('-C', normpath, 'pull', 'origin', rev)
-    else:
-        git('clone', '--reference', reference, repo, os.path.normpath(path))
+    if type == 'git':
+        name = repo.split('/')[-1]
+        if name.endswith(".git"):
+            name = name[:-4]
+
+        reference = os.path.join(cache, name + ".reference")
+        if not os.path.isdir(reference):
+            git('clone', '--mirror', repo, reference)
+
+        normpath = os.path.normpath(path)
+        if sparse_paths:
+            os.mkdir(normpath)
+            git('-C', normpath, 'init')
+            git('-C', normpath, 'config', 'core.sparseCheckout', 'true')
+            git('-C', normpath, 'remote', 'add', '-f', 'origin', reference)
+
+            with open("%s/%s/.git/info/sparse-checkout" % (os.getcwd(), normpath), 'w') as fd:
+                fd.writelines(sparse_paths)
+            with open("%s/%s/.git/objects/info/alternates" % (os.getcwd(), normpath), 'w') as fd:
+                fd.write("%s/objects" % reference)
+
+            # We use directly the revision requested here in order to respect,
+            # that not all repos have `master` as their default branch
+            git('-C', normpath, 'pull', 'origin', rev)
+        else:
+            git('clone', '--reference', reference, repo, os.path.normpath(path))
+    elif type == 'git-svn':
+        # just the preperation for the svn deep clone / checkout here (clone will be made in update function to simplify source.py).
+        os.makedirs(path)
```

Comment:
I would prefer it if we left the default cause flat rather than indent dozens of lines of code in each affected function:

```
if type == 'git-svn':
    # (new logic)
    return

assert type == `git`
# (current logic)
```

This should also make the diff much smaller.

Reply:
I did adapt the code regarding your advice.